### PR TITLE
Support per collection properties with Solr

### PR DESCRIPTION
**Pull request by pklinef**
#### What does this PR do?
Makes it possible to override per collection or by default the number of shards, replication factor, and allowed shards per node when collections are created.  The number of shards for the catalog collection is now 6 shards instead of 2 shards.  All other collections still have 2 shards by default.

#### Who is reviewing it? 

@lamhuy 
@derekwilhelm 

#### Select relevant component teams: 

@codice/solr 

#### Ask 2 committers to review/merge the PR and tag them here.

@rzwiefel
@bdeining

#### How should this be tested?

Go to http://localhost:8994/solr/#/catalog/collection-overview
Verify that the catalog collection has 6 shards.

Go to http://localhost:8994/solr/#/alerts/collection-overview
Verify that the alerts collection has 2 shards.

You can verify the values under `etc/solr/configsets/default/collection.properties` and `etc/solr/configsets/catalog/collection.properties`.

#### What are the relevant tickets?
Fixes: n/a

#### Checklist:
- [ n/a ] Documentation Updated
- [ n/a ] Update / Add Threat Dragon models
- [x] Update / Add Unit Tests
- [ n/a ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by pklinef** [comment]
build now

File: platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/CollectionConfig.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    this(collection, Paths.get(System.getProperty("ddf.home", ""), "etc/solr/configsets"));
+  }
+
+  public CollectionConfig(String collection, Path collectionConfigs) {
+    File matchingConfig =
+        collectionConfigs.resolve(Paths.get(collection, "collection.properties")).toFile();
+    Properties collectionProps = new Properties();
+    if (matchingConfig.exists()) {
+      try {
+        collectionProps.load(new FileInputStream(matchingConfig));
```
Comment (rzwiefel) [inline platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/CollectionConfig.java:46]: ❓ Should we be creating this `FileInputStream` with a try with resources? `Properties::load` states 
>  The specified stream remains open after this method returns.

Comment (pklinef) [reply platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/CollectionConfig.java]: Oops. Yes, I forgot to clean this up. Fixed.

**Comment by pklinef** [comment]
build now

File: platform/solr/solr-factory-impl/src/test/java/org/codice/solr/factory/impl/CollectionConfigTest.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class CollectionConfigTest {
```
Comment (bdeining) [inline platform/solr/solr-factory-impl/src/test/java/org/codice/solr/factory/impl/CollectionConfigTest.java:27]: do the coverage numbers change with this addition?

Comment (pklinef) [reply platform/solr/solr-factory-impl/src/test/java/org/codice/solr/factory/impl/CollectionConfigTest.java:27]: Maybe but I suspect it is probably much smaller than lenient variance.

**Pull request merged**